### DISCUSSION TITLE
Validation when using Troposphere based templates

### DISCUSSION THREAD

**Discussion by jerrygb** [cloudreach/sceptre#906] [Q&A]
Hi,

Is there a mechanism to pass parameters to the Troposphere templates as opposed to only `sceptre_user_data`.

As per the documentation, we see the function signature as follows,

```
from troposphere import Template
from troposphere.ec2 import VPC

def vpc(sceptre_user_data):
    """AWS VPC CloudFormationTemplate"""
    template = Template()
    template.add_resource(VPC(
            "VirtualPrivateCloud",
            CidrBlock=sceptre_user_data["cidr_block"]
        ))
    return template.to_yaml()

def sceptre_handler(sceptre_user_data):
    return vpc(sceptre_user_data)
```

We seem to have a basic use-case to validate a python based template.

```
template_path: vpc/subnets.py

stack_tags: 
  Environment: {{ command_path.0 }}

sceptre_user_data:
  VPCID: !stack_output {{ command_path.0 }}/{{ region }}/vpc/vpc.yaml::VPCID
  VPCCIDR: !stack_output {{ command_path.0 }}/{{ region }}/vpc/vpc.yaml::VPCCIDR
```

The _problem_ is that, in case, we `validate` this template without VPC, sceptre tries to execute the `stack_output` resolver and hence ends up with an error `Stack does not exist`.

We wanted to know if there is a way to pass parameters as well to the `Template` object.

May I please know if there a reason why we do not mock parameters and pass it in the following code, mimicking a similar behavior as yaml/jinja,
https://github.com/Sceptre/sceptre/blob/master/sceptre/stack.py#L269

Additionally, it would be great if we can have a _validation_ flag passed in. Not sure if it avoided due to a design concern.

Thanks.

**Comment by zaro0508** [comment]
Sceptre does allow passing in parameters with [Var](https://sceptre.cloudreach.com/2.3.0/docs/stack_group_config.html#var)  however that only overrides variables in config files.  Would that help your use case @jerrygb ?

**Comment by ngfgrant** [comment]
@jerrygb did Zaro's suggestion help you move forward?
